### DISCUSSION TITLE
Handling Exceptions Globally on Different Platforms

### DISCUSSION THREAD

**Discussion by abenadar** [unoplatform/uno#9202] [Q&A]
I am looking for some way to globally intercept any unhandled exceptions for analytic purposes and providing proper feedback to the user before crashing.

I configured a fallback for unhandled/unobserved exceptions equivalent to the UWP way (which for whatever reason doesn't even seem to work with the Windows head, might be related to [this ](https://github.com/microsoft/microsoft-ui-xaml/issues/5221) issue).

```
            UnhandledException += OnUnhandledException;

            TaskScheduler.UnobservedTaskException += OnUnobservedException;
```
These handlers do not seem to ever be hit on iOS/Android whatsoever either . I looked for workarounds and found this [polyfill](https://stackoverflow.com/questions/39503390/global-exception-handling-in-xamarin-cross-platform), which doesn't seem to be easily portable to Uno as is. 

I wonder if anyone has found any suggestion on how this could be solved in Uno, or even a ready-to-go solution. I fear the lack of options here to be slowly becoming a showstopper.

**Comment by spouliot** [comment]
For Xamarin.iOS and Xamarin.macOS _legacy_ (pre dotnet) SDK there are build options you can set that will affect how exceptions are handled. Note that this can have an effect on [runtime performance](https://github.com/xamarin/xamarin-macios/issues/14812#issuecomment-1108152654).

More details about the options @ https://docs.microsoft.com/en-us/xamarin/ios/platform/exception-marshaling

For the newer the iOS/macOS net6 SDK the [default is different](https://github.com/xamarin/xamarin-macios/issues/10574) (enabled) and, for macOS (CoreCLR) it cannot be disabled.

Still you might want to use a "crash reporter" tool/library (many exists) as they will be able to trap and report back many types of native crashes (not just managed exceptions).

**Reply by abenadar** [reply]
Thanks for these insights.

Though the thoughts on exception marshalling are helpful they don't seem to quite complement my initial issue as I don't even get as far as intercepting the exceptions.
Considering the changes in .NET 6 I tried porting the approach to .net6-Android/iOS hoping the behaviour might change, but alas the handlers are still not invoked.

Is my solution plagued by an error on my side or is it plainly bugged out (on Microsoft's side and/or within Uno's implementation).

Crash reporters like the on offered by TestFlight are nice to have but don't offer the in-depth flexibility I require for early testing builds.
